grype/vulnerability_matcher.go

```diff
@@ -122,7 +125,7 @@ func (m *VulnerabilityMatcher) searchDBForMatches(
 	release *linux.Release,
 	packages []pkg.Package,
 	progressMonitor *monitorWriter,
-) match.Matches {
+) (match.Matches, error) {
```

Comment:
Does surfacing this error here create a new case where grype can now fail for the following types when we cannot indexFalsePositivesByLocation?
```
`d.Type == distro.Wolfi || d.Type == distro.Chainguard || d.Type == distro.Alpine)`
```

Should we log the error and continue executing grype as we do on main now, or is this a good spot to add this new `grype exits on fail` condition?

Reply:
That's a good question. I defer to you and @wagoodman here... (Fun fact: we made this particular change standing over the same laptop at ALX!)

I got the impression from Alex that it felt like time to add the `error` return parameter at this point, but I'm not sure where else we'd want to return non-nil errors now.

Let me know what change you'd like me to make, if any — more than happy to.

Reply:
this is a pretty fundamental function -- if there really is an error condition here it could lead to missed results, which should always return 1 on the CLI (thus we should not log and continue).